src/main/com/diana/restaurant/queue/QueueManager.java

```diff
@@ -0,0 +1,54 @@
+package com.diana.restaurant.queue;
+
+import com.diana.restaurant.entities.EntityPriorityForQueue;
+import com.diana.restaurant.entities.Order;
+import com.diana.restaurant.enums.Queues;
+import com.diana.restaurant.exceptions.queueManager.QueueManagerKeyNotFoundException;
+import com.diana.restaurant.exceptions.services.NullEntityException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+public class QueueManager {
+  private static QueueManager instance;
+  private final Map<String, Queue> instanceMap = new HashMap<>();
```

Comment:
I  think you may want to handle a max capacity per queue.

Reply:
I added the class QueueManagerUtil with the variable max capacity, and added a condition when wanting to enQueue an item. If the queue size it's already at it's max capacity, it will throw an exception